File: sqlbuilder/sqlbuilder/Column.java
package com.clickability.cms.dataaccess.sqlbuilder;

public class Column implements Expression {
	
	public final String name;
	public final Resource resource;
	
	public Column(String name) {
		this.name = name;
		this.resource = null;
	}

	public Column(Resource resource, String name) {
		this.name = name;
		this.resource = resource;
	}

	@Override
	public String get() {
		return this.resource == null ?
			this.name : 
			String.format("%s.%s", this.resource.alias(), this.name);
	}

}

File: queryBuilder/src/main/kotlin/queryBuilder/com/inunison/lib/queryBuilder/Column.kt
package com.clickability.cms.dataaccess.sqlbuilder

class Column : Expression {

    val name: String
    val resource: Resource?

    constructor(name: String) {
        this.name = name
        this.resource = null
    }

    constructor(resource: Resource, name: String) {
        this.name = name
        this.resource = resource
    }

    @Override
    fun get(): String {
        return if (this.resource == null)
            this.name
        else
            String.format("%s.%s", this.resource!!.alias(), this.name)
    }

}
